Disable chronod when booting simulated devices for testing.

bls1999 opened the pull request:
#### 72c9c0bd54a86cce02e64a1b9fe8b6c3e57851b4
<pre>
Disable chronod when booting simulated devices for testing.
<a href="https://rdar.apple.com/130111123">rdar://130111123</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=275623">https://bugs.webkit.org/show_bug.cgi?id=275623</a>

Reviewed by Elliott Williams.

We&apos;re observing abnormally large memory usage from widgets-related processes on the
iOS simulator. To combat this during investigation, we should unload `chronod` at
simulator startup.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDeviceManager.Runtime.__init__): Add runtime.root (path to simulator runtime root).
(SimulatedDeviceManager._create_device_with_runtime): Pass runtime root path to device init method.
(SimulatedDevice.__init__): Add device.runtime_root (path to simulator runtime root).
* Tools/Scripts/webkitpy/xcode/simulated_device_unittest.py: Update mock data to include runtimeRoot.

Canonical link: <a href="https://commits.webkit.org/280818@main">https://commits.webkit.org/280818@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c6f99b3e62391fcc883cee502158b2a0cafd9562

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57741 "Failed to checkout and rebase branch from PR 29939") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37069 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10217 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61363 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8186 "Built successfully") 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/59869 "Failed to checkout and rebase branch from PR 29939") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44705 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8374 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/46784 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/5805 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59771 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/34764 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/49895 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/27609 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/57267 "Failed to checkout and rebase branch from PR 29939") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/31534 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/7187 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7190 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/53486 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/7458 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/63045 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1655 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/7531 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/63045 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1661 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49906 "Build is in progress. Recent messages:") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/63045 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/88/builds/1400 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8605 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/32898 "Built successfully") | | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/33984 "Failed to checkout and rebase branch from PR 29939") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35068 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33729 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Tools/Scripts/webkitpy/xcode/simulated_device.py:
```diff
@@ -536,20 +538,23 @@ class DeviceState:
 
-        self.environment_extras = []
+        self.environment_extras = [
+            [SimulatedDeviceManager.xcrun, 'simctl', 'spawn', self.udid, 'launchctl', 'unload', '-w', f'{runtime_root}/System/Library/LaunchDaemons/com.apple.chronod.plist'],  # FIXME: webkit.org/b/275623
```
Worth noting that this should persist between reboots of the simulator device. But I think it's fine to unconditionally run it anyways.

bls1999 replied:
All simulators are _removed_ at tear down, so it unfortunately doesn't persist.

emw-apple commented on Tools/Scripts/webkitpy/xcode/simulated_device.py:
```diff
@@ -536,20 +538,23 @@ class DeviceState:
 
-        self.environment_extras = []
+        self.environment_extras = [
+            [SimulatedDeviceManager.xcrun, 'simctl', 'spawn', self.udid, 'launchctl', 'unload', '-w', f'{runtime_root}/System/Library/LaunchDaemons/com.apple.chronod.plist'],  # FIXME: webkit.org/b/275623
```
It may be clearer to link to the radar tracking the widgets bug, since the WebKit bug you linked will get closed as fixed once you land this.

bls1999 committed:
Disable chronod when booting simulated devices for testing.
rdar://130111123
https://bugs.webkit.org/show_bug.cgi?id=275623

Reviewed by Elliott Williams.

We're observing abnormally large memory usage from widgets-related processes on the
iOS simulator. To combat this during investigation, we should unload `chronod` at
simulator startup.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDeviceManager.Runtime.__init__): Add runtime.root (path to simulator runtime root).
(SimulatedDeviceManager._create_device_with_runtime): Pass runtime root path to device init method.
(SimulatedDevice.__init__): Add device.runtime_root (path to simulator runtime root).
* Tools/Scripts/webkitpy/xcode/simulated_device_unittest.py: Update mock data to include runtimeRoot.

Canonical link: https://commits.webkit.org/280818@main
